pkg/catalog/manager/traverse.go

```diff
@@ -231,15 +225,16 @@ func (m *Manager) traverseAndUpdate(repoPath, commit string, cmt *CatalogInfo, t
 	}
 	// delete non-existing templates
 	for _, toDelete := range toDeleteChart {
-		logrus.Debugf("Deleting template %s and its associated templateVersion", toDelete)
+		logrus.Debugf("Deleting template %s and its associated templateVersion in namespace %s", toDelete, templateNamespace)
 		if err := m.deleteChart(toDelete, templateNamespace); err != nil {
-			return err
+			logrus.Error(err)
```

Comment:
is there a reason to not return this error?

Reply:
Good catch! 

I think there might be a bug in norman, in combination with the traverse code in master where returning an error here causes an endless loop of unrecoverable errors. In master code when we refresh the catalog we don't check if the catalog has been removed, so disabling a catalog while it is updating may cause some weird values to be used. 

I think if we attempt to delete a namespaced resource without a namespace the resource became updated but not removed, and then future attempts to remove it would cause the continuous errors. I didn't dig super far into the conditions or root cause of what triggered the loop.